packages/flutter_tools/bin/xcode_backend.sh

```diff
@@ -257,11 +257,10 @@ ThinFramework() {
 }
 
 ThinAppFrameworks() {
-  local app_path="${TARGET_BUILD_DIR}/${WRAPPER_NAME}"
-  local frameworks_dir="${app_path}/Frameworks"
+  local xcode_frameworks_dir="${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}"
```

Comment:
This is the actual fix.
Instead of `find`ing and thinning all `*.framework`s in `Runner.app`, it now only does so in the `Runner.app/Frameworks` directory.